From: Eugenia Emantayev <eugenia@mellanox.com>
Subject: [PATCH] BACKPORT: add backport for timestamping ethtool option

Signed-off-by: Eugenia Emantayev <eugenia@mellanox.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
index xxxxxxx..xxxxxxx xxxxxx
--- a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
@@ -1585,6 +1585,7 @@ out:
 	return err;
 }
 
+#ifdef CONFIG_TIMESTAMP_ETHTOOL
 static int mlx4_en_get_ts_info(struct net_device *dev,
 			       struct ethtool_ts_info *info)
 {
@@ -1613,6 +1614,7 @@ static int mlx4_en_get_ts_info(struct net_device *dev,
 
 	return ret;
 }
+#endif
 
 #if ((LINUX_VERSION_CODE > KERNEL_VERSION(2,6,35)) && (LINUX_VERSION_CODE < KERNEL_VERSION(2,6,39)))
 int mlx4_en_set_flags(struct net_device *dev, u32 data)
@@ -1685,7 +1687,9 @@ const struct ethtool_ops_ext mlx4_en_ethtool_ops_ext = {
 	.get_channels = mlx4_en_get_channels,
 	.set_channels = mlx4_en_set_channels,
 #endif
+#ifdef CONFIG_TIMESTAMP_ETHTOOL
 	.get_ts_info = mlx4_en_get_ts_info,
+#endif
 };
 
 
